### DISCUSSION TITLE
Pagination?

### DISCUSSION THREAD

**Discussion by bluz71** [beercss/beercss#383] [Q&A]
I am evaluating Beer CSS, and it is indeed most impressive.

However, I am do not see a solution for the *Pagination* pattern.

How are folks styling Pagination with Beer CSS?

Examples of Pagination in other CSS frameworks:

- [Material UI](https://mui.com/material-ui/react-pagination)
- [Tailwind UI](https://tailwindui.com/components/application-ui/navigation/pagination)

Best regards.

**Comment by laurentpayot** [comment]
I could not find MD3 specs for pagination, only MD2 specs: https://m2.material.io/components/data-tables#anatomy

I use infinite scrolling to avoid pagination. Maybe that’s implicitly recommended by this absence in the specs?

**Comment by coffeeandwork** [comment]
Hard to imagine replacing pagination with infinite scroll across the board. 

The following products (some of them, from google), don't seem to have given up pagination:
- gmail
- google search 
- https://console.cloud.google.com/apis/dashboard
- github (issues, discussion, pull requests)

on the other hand, the following products don't seem to use much pagination (mostly just infinite scroll):
- meta
- x.com
- outlook

Perhaps if beercss ever decided to go MD3++, paginator could be a good candidate to consider adding ... just thinking out loud here...

**Reply by acroyear** [reply]
My company's design language also removed pagination in favor of infinite scroll. The main reason is that beyond a certain amount of data, it is more expected that the user would search/filter instead of scrolling to find something.

Pagination is actually more annoying because you don't really know what page something might be on. Take gmail as one example - if I need an email from March 4th...what page is it on? I don't know. I have to skip a few pages and see where i'm at, and of course, the data is variable and unpredictable. One page of data may have only one day's emails, another page may have 2 weeks worth because I deleted everything at that time after reading it. It is guesswork.

That's why pagination as a UX design pattern is being retired. A proper app should improve its search and filter.

The google products that still have pagination have had it since they were created. None are Material-You / 3 / Expressive compliant.

Google search on Mobile is Material-You compliant (not yet Expressive) - it has infinite scroll, though explicit rather than automatic: you have to click a button to load more results. Gmail on the phone also is infinite scroll now as well.

So since there's no Material Expressive pagination design, you're free to just make your own. Maybe start with https://github.com/flowforfrank/pagination, remove any style css (keeping structure) and then add beercss classes to it. MIT license, so have at it. But I don't think beercss should otherwise take responsibility for building and maintaining one.

Another pagination option is the slider with the tooltip option. put that, plus buttons on each end for 1 and Last (<< and >> icons usually), and then maybe a number input to type the page number exactly and it always shows the current page otherwise as set by the slider. So just 4 components (button, slider, input, button) and a little javascript to connect the slider and the input. I could probably bundle that in a vue or web-component in about half an hour.
